misc/git/hooks/unused

```diff
@@ -18,7 +18,9 @@ gofiles=$(git diff --cached --name-only --diff-filter=ACM | grep '^go/.*\.go$' |
 if [ "$gofiles" = "" ]; then
   exit 0
 fi
-gopackages=$(echo $gofiles | xargs dirname | sort -u)
+gopackages=$(for i in $gofiles; do
+  dirname $i
+done | sort -u)
```

Comment:
The other choice:
xargs -n 1 dirname
only passes one parameter at a time.

The original line is weird, could just have been:
dirname $gofiles | sort -u
?

Anyway, I like your new version better :)

Reply:
There's an limit to the number of characters for the total arguments string. xargs helps in situations like that, so it made sense to have it in the first place :)

I actually like the `-n1` idea better because the code stays a one-liner this way. I've updated the commit.

Reply:
And 'echo $gofiles' is not subject to the number of characters limit, because it's a bash built-in command, and it's never forking /bin/echo?

(just curious here :) )

Reply:
bash built-ins do not seem to be affected :)

https://www.in-ulm.de/~mascheck/various/argmax/

> Thus the following commands work smoothly, because instead of handing over too many arguments to a new process, 
> they only make use of a shell built-in (echo) or iterate over the arguments with a control structure (for loop):
> 
>     /dir-with-many-files$ echo * | wc -c